### PR TITLE
Extend DiffMorph to make it navigable accross differences

### PR DESCRIPTION
**Pull request by sonibla**
Hello there

This PR makes it possible to add buttons to easily navigate in the text comparator

navigableDiffPresenter

Defult behavior of DiffMorph unchanged (by default no buttons will show up)

This PR only modifies the Morphic classes, for the Spec one I did another PR on Spec repository: https://github.com/pharo-spec/Spec/pull/1305

### CODE REVIEW

**Comment by sonibla** [comment]
Adaptation of @rolandbernard 's work, credits to him

File: src/Tool-Diff/DiffMorph.class.st [hunk 1]
```diff
@@ -259,6 +285,86 @@ DiffMorph >> beWrapped: aBoolean [
...
+		  listCentering: #bottomRight;
+		  yourself
+]
+
+{ #category : #actions }
+DiffMorph >> calculateAllDifferences [
+
+	| diff si di |
+	si := 1.
+	di := 1.
```
Comment (demarey) [inline src/Tool-Diff/DiffMorph.class.st]: would it be possible to use meaningful names for `si`and `di`?

File: src/Tool-Diff/DiffMorph.class.st [hunk 2]
```diff
@@ -259,6 +285,86 @@ DiffMorph >> beWrapped: aBoolean [
...
+	diff := OrderedCollection new.
+	self joinMappings do: [ :join | 
+		join type = #match ifFalse: [ 
+			join type = #modification
+				ifTrue: [ 
+					| inlineDiff sj dj |
+					sj := si.
```
Comment (demarey) [inline src/Tool-Diff/DiffMorph.class.st]: would it be possible to use meaningful names for `sj` and `dj`?

File: src/Tool-Diff/DiffMorph.class.st [hunk 3]
```diff
@@ -259,6 +285,86 @@ DiffMorph >> beWrapped: aBoolean [
...
+					dj := di.
+					inlineDiff := (InlineTextDiffBuilder
+						               from: join src text
+						               to: join dst text) buildPatchSequence 
+						              groupByRuns: [ :e | e key = #match ].
+					inlineDiff do: [ :c | 
+						c first key = #match
```
Comment (demarey) [inline src/Tool-Diff/DiffMorph.class.st]: would it be possible to use meaningful name for `c`?

File: src/Tool-Diff/DiffMorph.class.st [hunk 4]
```diff
@@ -259,6 +285,86 @@ DiffMorph >> beWrapped: aBoolean [
...
+							ifTrue: [ 
+								c do: [ :s |
```
Comment (demarey) [inline src/Tool-Diff/DiffMorph.class.st]: would it be possible to use meaningful names for `s`?

File: src/Tool-Diff/DiffMorph.class.st [hunk 5]
```diff
@@ -259,6 +285,86 @@ DiffMorph >> beWrapped: aBoolean [
...
+						c first key = #match
+							ifTrue: [ 
+								c do: [ :s | 
+									sj := sj + s value size.
+									dj := dj + s value size ] ]
+							ifFalse: [ 
+								| sk dk |
+								sk := sj.
+								dk := dj.
+								c do: [ :s |
```
Comment (demarey) [inline src/Tool-Diff/DiffMorph.class.st]: would it be possible to use meaningful names for `sk` and `dk`?

**Review by demarey — CHANGES_REQUESTED** [review]

**Comment by Ducasse** [comment]
Ping

**Comment by sonibla** [comment]
@demarey thanks for your review!
I refactored calculateAllDifferences, it's now easier to read

**Pull request merged**